src/Container/AgFeatureGrid/AgFeatureGrid.jsx

```diff
@@ -0,0 +1,1021 @@
+   * @return {Object} The row candidate.
+   */
+  getRowFromFeatureId = id => {
+    const {
+      grid
+    } = this.state;
+
+    let rowNode;
+
+    grid.api.forEachNode(node => {
+      if (node.data.key === id) {
+        rowNode = node;
+      }
+    });
+
+    return rowNode;
+  }
+
+  /**
+   * Returns the currently selected row keys.
+   *
+   * @return {Number[]} An array with the selected row keys.
+   */
+  getSelectedRowKeys = () => {
+    const {
+      grid
+    } = this.state;
+
+    const selectedRows = grid.api.getSelectedRows();
+
+    return selectedRows.map(row => row.key);
+  }
+
+  /**
+   * Called on row click and zooms the the corresponding feature's extent.
+   *
+   * @param {Object} row The clicked row.
+   */
+  onRowClick = evt => {
+    const {
+      onRowClick
+    } = this.props;
+
+    const row = evt.data;
+    const feature = this.getFeatureFromRowKey(row.key);
+
+    if (isFunction(onRowClick)) {
+      onRowClick(row, feature);
+    }
+
+    this.zoomToFeatures([feature]);
+  }
+
+  /**
+   * Called on row mouseover and hightlights the corresponding feature's
+   * geometry.
+   *
+   * @param {Object} row The highlighted row.
+   */
+  onRowMouseOver = evt => {
+    const {
+      onRowMouseOver
+    } = this.props;
+
+    const row = evt.data;
+    const feature = this.getFeatureFromRowKey(row.key);
+
+    if (isFunction(onRowMouseOver)) {
+      onRowMouseOver(row, feature);
+    }
+
+    this.highlightFeatures([feature]);
+  }
+
+  /**
+   * Called on mouseout and unhightlights any highlighted feature.
+   *
+   * @param {Object} row The unhighlighted row.
+   */
+  onRowMouseOut = evt => {
+    const {
+      onRowMouseOut
+    } = this.props;
+
+    const row = evt.data;
+    const feature = this.getFeatureFromRowKey(row.key);
+
+    if (isFunction(onRowMouseOut)) {
+      onRowMouseOut(row, feature);
+    }
+
+    this.unhighlightFeatures([feature]);
+  }
+
+  /**
+   * Fits the map's view to the extent of the passed features.
+   *
+   * @param {ol.Feature[]} features The features to zoom to.
+   */
+  zoomToFeatures = features => {
+    const {
+      map
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    let featGeometries = [];
+    features.forEach(feature => {
+      featGeometries.push(feature.getGeometry());
+    });
+
+    if (featGeometries.length > 0) {
+      const geomCollection = new OlGeomGeometryCollection(featGeometries);
+      map.getView().fit(geomCollection.getExtent());
+    }
+  }
+
+  /**
+   * Highlights the given features in the map.
+   *
+   * @param {ol.Feature[]} highlightFeatures The features to highlight.
+   */
+  highlightFeatures = highlightFeatures => {
+    const {
+      map,
+      highlightStyle
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    highlightFeatures.forEach(feature => feature.setStyle(highlightStyle));
```

Comment:
Plz make use of a `highlightLayer` to avoid setting the style directly on the feature.

Reply:
True. That code has been taken over from the original `FeatureGrid`. I've opened #594 to refactor/fix this.